### PR TITLE
feat: Disable legacy rate limits when unused

### PR DESCRIPTION
**Pull request by dcramer**
These have been superceded by per-key rate limits.

### CODE REVIEW

**Comment by dcramer** [comment]
This is blocked by @getsentry/cops making some changes

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by dcramer** [comment]
For anyone paying attention, we're going to wait to land this until we improve the spike detection feature

**Comment by dcramer** [comment]
@getsentry/cops i'm putting a deadline of two weeks to be content w/ this for customers and then I'm asking EPD to push this through

File: tests/sentry/api/endpoints/test_organization_details.py [hunk 1]
```diff
@@ -227,6 +211,52 @@ def test_various_options(self):
...
+                'organization_slug': org.slug,
+            }
+        )
+        response = self.client.put(
+            url,
+            data={
+                'accountRateLimit': 1000,
+            }
+        )
+        assert response.status_code == 400, response.content
```
Comment (billyvg) [inline tests/sentry/api/endpoints/test_organization_details.py]: Why does this return a 400?

Comment (dcramer) [reply tests/sentry/api/endpoints/test_organization_details.py]: you cannot configure this option if you haven't done so previously, so attempting to set it returns an error

**Comment by dcramer** [comment]
@lynnagara @billyvg if one of yall mind reviewing/approving this should be good to go

**Review by lynnagara — APPROVED** [review]

**Pull request merged**